File: src/main/java/net/alephdev/calendar/models/Release.java
package net.alephdev.calendar.models;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import lombok.Data;

@Data
@Entity
@Table(name = "Releases")
@SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
public class Release {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Integer id;

  @Column(nullable = false)
  private String version;

  @Column(nullable = false)
  private LocalDate releaseDate;

  private String description;

  @ManyToOne
  @JoinColumn(name = "sprint_id", foreignKey = @ForeignKey(name = "fk_releases_sprint_id"))
  private Sprint sprint;
}

File: angular/src/app/models/release.ts
import {Sprint} from "./sprint";

/**
 * Модель релиза продукта.
 *
 * Связан со спринтом и содержит версию и дату релиза.
 *
 * @property id - Уникальный идентификатор релиза
 * @property version - Версия релиза (например, 1.0.0)
 * @property releaseDate - Дата релиза в ISO формате
 * @property description - Описание релиза
 * @property sprint - Спринт, к которому относится релиз
 */
export interface Release {
  id: number;
  version: string;
  releaseDate: string; // Date in ISO format
  description: string | null;
  sprint: Sprint | null;
}
